File: Java/src/main/java/Solution101_200/Solution168.java
package Solution101_200;

/**
 * @author windows
 */
public class Solution168 {
    public String convertToTitle(int columnNumber) {
        StringBuilder sb = new StringBuilder();
        while(columnNumber > 26){
            int a = columnNumber % 26;
            if (a == 0){
                a = 26;
                columnNumber -= 26;
            }
            sb.append((char)(a + 'A' - 1));
            columnNumber /= 26;
        }
        sb.append((char)(columnNumber + 'A' - 1));
        return sb.reverse().toString();
    }


    public static void main(String[] args) {
        Solution168 solution168 = new Solution168();
        System.out.println(solution168.convertToTitle(52));
    }
}

File: C++/src/Solution101_200/Solution168.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution168.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string Solution168::convertToTitle(int columnNumber) {
    string res = "";
    while (columnNumber > 0){
        columnNumber--;
        int remain = columnNumber % 26;
        res = (char)('A' + remain) + res;
        columnNumber /= 26;
    }
    return res;
}
